pkg/util/encoding/encoding.go

```diff
@@ -3127,6 +3127,9 @@ func DecodeUUIDValue(b []byte) (remaining []byte, u uuid.UUID, err error) {
 
 // DecodeUntaggedUUIDValue decodes a value encoded by EncodeUntaggedUUIDValue.
 func DecodeUntaggedUUIDValue(b []byte) (remaining []byte, u uuid.UUID, err error) {
+	if len(b) < uuidValueEncodedLength {
+		return b, uuid.UUID{}, errors.Errorf("invalid uuid length of %d", len(b))
```

Comment:
Should this be an assertion error instead?

Reply:
Most of these decode functions use regular errors (including `decodeValueTypeAssert`). I think for now an assertion error is overkill.
<!-- Reviewable comment -OCZQG9TDNq8z12x2Z8U:bdrx5br -->
<!-- Sent from Reviewable.io -->